### PR TITLE
TP-1147 Add remote partials and js for submission flag/unflag

### PR DESCRIPTION
**Pull request by aktfrikshun**
TP-1147 Add remote partials and js for submission flag/unflag

### CODE REVIEW

File: spec/features/admin/submissions_spec.rb [hunk 1]
```diff
@@ -47,7 +47,6 @@
             end
 
             it "successfully flags Submission" do
-              expect(page).to have_content("Response #{submission.id} was successfully flagged.")
```
Comment (ryanwoldatwork) [inline spec/features/admin/submissions_spec.rb:50]: is there a way to assert the flag happened? maybe by flag color/status?

File: app/views/admin/submissions/flag.js.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+$("#flag-submission-<%= @submission.id %>").parent().html("<%= j (render partial: 'flag', :locals => {submission: @submission}) %>")
```
Comment (ryanwoldatwork) [inline app/views/admin/submissions/flag.js.erb:1]: can the element be targeted by submission id, then within the submission, find the flag td/element?

File: spec/features/admin/submissions_spec.rb (hunk 1 above)
Comment (aktfrikshun) [reply spec/features/admin/submissions_spec.rb:50]: Page having content Flagged asserts the flag happened

File: app/views/admin/submissions/flag.js.erb (hunk 2 above)
Comment (aktfrikshun) [reply app/views/admin/submissions/flag.js.erb:1]: Sure, I can do that

Comment (aktfrikshun) [reply app/views/admin/submissions/flag.js.erb]: @ryanwoldatwork  Done

**Pull request merged**